database/misc/mongodb_replication.py

```diff
@@ -0,0 +1,410 @@
+            for member in cfg['members']:
+                if host_name in member['host']:
+                    cfg['members'].remove(member)
+                else:
+                    fail_msg = "couldn't find member with hostname: {0} in replica set members list".format(host_name)
+                    module.fail_json(msg=fail_msg)
+        except (OperationFailure, AutoReconnect), e:
+            timeout = timeout - 5
+            if timeout <= 0:
+                module.fail_json(msg='reached timeout while waiting for rs.reconfig(): %s' % str(e))
+            time.sleep(5)
+
+def load_mongocnf():
+    config = ConfigParser.RawConfigParser()
+    mongocnf = os.path.expanduser('~/.mongodb.cnf')
+
+    try:
+        config.readfp(open(mongocnf))
+        creds = dict(
+          user=config.get('client', 'user'),
+          password=config.get('client', 'pass')
+        )
+    except (ConfigParser.NoOptionError, IOError):
+        return False
+
+    return creds
+
+def wait_for_ok_and_master(module, client, timeout = 60):
+    while True:
+        status = client.admin.command('replSetGetStatus', check=False)
+        if status['ok'] == 1 and status['myState'] == 1:
+            return
+
+        timeout = timeout - 1
+        if timeout == 0:
+            module.fail_json(msg='reached timeout while waiting for rs.status() to become ok=1')
+
+        time.sleep(1)
+
+def authenticate(client, login_user, login_password):
+    if login_user is None and login_password is None:
+        mongocnf_creds = load_mongocnf()
+        if mongocnf_creds is not False:
+            login_user = mongocnf_creds['user']
+            login_password = mongocnf_creds['password']
+        elif login_password is None and login_user is not None:
+            module.fail_json(msg='when supplying login arguments, both login_user and login_password must be provided')
+
+    if login_user is not None and login_password is not None:
+        client.admin.authenticate(login_user, login_password)
+
+# =========================================
+# Module execution.
+#
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            login_user=dict(default=None),
+            login_password=dict(default=None),
+            login_host=dict(default='localhost'),
+            login_port=dict(default='27017'),
+            replica_set=dict(default=None),
+            host_name=dict(default='localhost'),
+            host_port=dict(default='27017'),
+            host_type=dict(default='replica', choices=['replica','arbiter']),
+            ssl=dict(default='false'),
+            build_indexes = dict(type='bool', default='yes'),
+            hidden = dict(type='bool', default='no'),
+            priority = dict(default='1.0'),
+            slave_delay = dict(type='int', default='0'),
+            votes = dict(type='int', default='1'),
+            state=dict(default='present', choices=['absent', 'present']),
+        )
```

Comment:
https://github.com/ansible/ansible-modules-extras/pull/265/files#diff-7848f393ef3ece46e90229b6499e39a7R308
User will see a message about it.